tests/end_to_end_tests/test_register_dataset_alias.py

```diff
@@ -41,3 +42,24 @@ def test_register_dataset_alias(dummy_file):
         assert i < 1
         assert getattr(r, "dataset.dataset_id") == d_id
         assert getattr(r, "dataset_alias.dataset_id") == d_id
+
+    # Try to reuse alias without supersede.  Should fail
+    a2_id = _insert_alias_entry(datareg.Registrar, "nice_dataset_name", d2_id)
+    assert a2_id is None
```

Comment:
Is there a benefit to this scenario not raising an exception. I can imagine just a print statement might get lost in a large script.

Reply:
I don't understand what you're proposing. The desired behavior is that the insert will fail, so the assert statement is asserting that.